tests/naive.py

```diff
@@ -2129,3 +2149,16 @@ def isconstant_func_stddev_threshold(a, w, quantile_threshold=0, stddev_threshol
             stddev_threshold = 0
 
     return sliding_stddev <= stddev_threshold
+
+
+def select_mpdist(P_ABBA, m, percentage, n_A, n_B):
```

Comment:
what is this? why is it called "select"? later, why do we compare core._mpdist with this function?

Reply:
I moved the following function

https://github.com/TDAmeritrade/stumpy/blob/814977f110403bcfd1629bcf1ba4f90938394bfe/tests/test_mpdist.py#L153-L166

to `tests/test_core.py`, and I just made slight changes. But before that, I would like to bring your attention to one thing here. I am not sure if this is a correct way of testing `custom_func` since the `custom_func` we are passing here is just for `percentage = 0.05`, and note that the naive version is being computed only based on `T_A`, `T_B`, and `m`. In other words, we are just testing the DEFAULT value of `custom_func`. 

---

I just moved `some_func` to `tests/naive.py` and tried to just rename it as `some_func` was a bit strange for me.

Reply:
>  I am not sure if this is a correct way of testing custom_func since the custom_func we are passing here is just for percentage = 0.05, and note that the naive version is being computed only based on T_A, T_B, and m. In other words, we are just testing the DEFAULT value of custom_func.

1. Since we don't actually use `k` anywhere in this test, we should remove `@pytest.mark.parametrize("k", k)` and update the function definition to `def test_mpdist_custom_func(T_A, T_B):` (note the removal of `k`)
2. The goal of this test is simply to see if `_mpdist` understands how to accept a custom function and to infer `k` from it. From what I can see, there is no default `percentage` value for `custom_func` but we are testing `custom_func` against the default `percentage` value of `_mpdist`, which is perfectly fine. Again, the goal is to simply check if the `custom_func`, which is a wrapped partial function, can produce the same outputs via `_mpdist` as the naive version when neither `k` nor `percentage` is provided as parameters to `_mpdist`. This is the most basic test but, I think, a reasonable one as-is

Reply:
> I just moved some_func to tests/naive.py and renamed it to select_mpdist since the name some_func was a bit strange for me. I think we are just using this function (i.e. some_func, which is now: select_mpdist) in one test function. So, we may do:

Thank you. Moving it is fine. Instead of doing a closure, I think simply renaming it from the current `select_mpdist` to `mpdist_custom_func` would be fine

Reply:
> 1. Since we don't actually use `k` anywhere in this test, we should remove `@pytest.mark.parametrize("k", k)` and update the function definition to `def test_mpdist_custom_func(T_A, T_B):` (note the removal of `k`)

Right! I will revise it. Thank you for catching it.

> 2. The goal of this test is simply to see if `_mpdist` understands how to accept a custom function and to infer `k` from it. From what I can see, there is no default `percentage` value for `custom_func` but we are testing `custom_func` against the default `percentage` value of `_mpdist`, which is perfectly fine. Again, the goal is to simply check if the `custom_func`, which is a wrapped partial function, can produce the same outputs via `_mpdist` as the naive version when neither `k` nor `percentage` is provided as parameters to `_mpdist`. This is the most basic test but, I think, a reasonable one as-is

Got it now. Thank you for your explanation!

Reply:
@seanlaw 

> 1. Since we don't actually use `k` anywhere in this test, we should remove `@pytest.mark.parametrize("k", k)` and update the function definition to `def test_mpdist_custom_func(T_A, T_B):` (note the removal of `k`)

I just noticed that I already removed it. Is there any other function that you wanted to bring my attention into?

Reply:
No, I think that was it. I haven't gone through `naive.py` and the unit tests yet